fdbserver/ClusterController.actor.cpp

```diff
@@ -802,6 +802,20 @@ class ClusterControllerData {
 			return false;
 		}
 
+		if(!clusterControllerProcessId.present()) {
+			return false;
+		}
+
+		auto ccWorker = id_worker.find(clusterControllerProcessId.get());
+		if(ccWorker == id_worker.end()) {
```

Comment:
If I understand correctly, the only way this can happen is if the cluster controller worker is detected as failed in `workerAvailabilityWatch`. If that is possible for the cluster controller process, is there a reason we aren't clearing `clusterControllerProcessId` at the same time (which would make this check unnecessary)?

There are some places that access `id_worker[clusterControllerProcessId]` without checking for its presence as far as I can tell, and if the worker has already been erased this is going to recreate it. That could result in some memory being lost to these entries, and it may also result in wrong behavior (I haven't really checked).